### PR TITLE
add --all-versions options to test runs

### PR DESCRIPTION
**Pull request by vogel**
Now it can be selected at runtime if tests are to be run againsts all
protocol versions or only against current.

Signed-off-by: Rafał Malinowski <rafal.przemyslaw.malinowski@gmail.com>

### CODE REVIEW

File: src/test/test.cpp [hunk 1]
```diff
@@ -151,15 +152,30 @@ test(int argc, char* const* argv, el::Level ll,
...
+    Session session{};
+    auto r =
+        session.applyCommandLine(argc, argv, Session::OnUnusedOptions::Ignore);
+    if (r != 0)
+        return r;
+
+    auto unusedTokens = session.unusedTokens();
+    using namespace Clara;
+    gTestAllVersions =
+        std::find_if(std::begin(unusedTokens), std::end(unusedTokens),
```
Comment (MonsieurNicolas) [inline src/test/test.cpp]: I'd rather have an explicit loop over `unusedToken` and error if we don't recognize one of them

Comment (vogel) [reply src/test/test.cpp]: Updated.

Note that Catch 2.0 has much better API for adding own command line arguments - https://github.com/catchorg/Catch2/blob/master/docs/own-main.md#adding-your-own-command-line-options

Comment (MonsieurNicolas) [reply src/test/test.cpp]: cool. Ah yes 2.0 seems promising, but it won't be cheap. I see that they have a bunch of breaking changes like removed `Catch::toString` . We can probably consider switching to it in a few months when they have hammered out the bugs.

**Comment by MonsieurNicolas** [comment]
r+ eaa4a6628474558bd2d6445e8089afeb2802efef

**Pull request merged**